src/lib/layer/layer-item/layer-item.component.ts

```diff
@@ -49,7 +52,14 @@ export class LayerItemComponent {
     this.layer.opacity = opacity / 100;
   }
 
-  constructor(private metadataService: MetadataService) { }
+  constructor(private cdRef: ChangeDetectorRef,
+              private metadataService: MetadataService,
+              private mapService: MapService) {
+
```

Comment:
Since layer-item is not necessarily binded to a map, the mapService shouldn't be used here.